SFML/SFML.Net
# Handle disposable objects propertly

# Description

Currently, SFML.Net has no checks for whenever an object is disposed (i.e. its CPointer is set to IntPtr.Zero), and will call methods and getters without complaining (before promptly crashing).

The main reasons why this is not great are:
1. Users will crash with a potentially unclear reason as to why (such as "Attempting to write into protected memory", i.e. the classic segfault)
2. Debugger sessions (such as VS) are unable to handle SFML's disposed objects correctly and will promptly crash and stop the debugging session upon attempting to view that object's properties in the Locals / Watch tabs.

Solving this issue isn't too hard, but adds checks across the entire codebase:
- Most implementations of ToString() have been changed so that they display that an object has been disposed instead of attempting to read properties for it
- All member function calls and properties that use CPointer / do CSFML calls have been guarded with a method that throws an ObjectDisposedException when trying to access those methods and properties.

# How to test?

Run each of the example projects and make sure they don't crash for normal objects (i.e. not disposed.)
Additionally, run code similar to this one in a debugger session to make sure disposed objects can be visualized correctly in Locals / Watch:

```cs
var disposables = new IDisposable[]
{
    GenerateSineWave(frequency: 440.0, volume: .25, seconds: 1),
    new Sound(GenerateSineWave(frequency: 440.0, volume: .25, seconds: 1)),
    new RenderWindow(new VideoMode(1280, 720), "Test"),
    new Window(new VideoMode(1280, 720), "Test2"),
    new Texture(128, 128),
    new RectangleShape(new Vector2f(128, 128)),
    new CircleShape(128, 128),
    new ConvexShape(32),
};

foreach (var obj in disposables)
{
    obj.Dispose();
    Console.WriteLine(obj.ToString());
}
```
